File: app/src/main/java/com/chad/baserecyclerviewadapterhelper/base/BaseBindingViewHolder.java
package com.chad.baserecyclerviewadapterhelper.base;

import android.databinding.ViewDataBinding;
import android.view.View;

import com.chad.library.adapter.base.BaseViewHolder;

/**
 * Date: 2017/5/11 14:42.
 * Email: tyshengsx@gmail.com
 * @author tysheng
 */

public class BaseBindingViewHolder<B extends ViewDataBinding> extends BaseViewHolder {
    private B mB;

    public BaseBindingViewHolder(View view) {
        super(view);
    }

    public B getBinding() {
        return mB;
    }

    public void setBinding(B b) {
        mB = b;
    }
}

File: kotlinapp/src/main/java/com/allen/kotlinapp/base/BaseBindingViewHolder.kt
package com.allen.kotlinapp.base

import android.databinding.ViewDataBinding
import android.view.View
import com.chad.library.adapter.base.BaseViewHolder

/**
 * 文 件 名: BaseBindingViewHolder
 * 创 建 人: Allen
 * 创建日期: 2017/6/13 15:05
 * 修改时间：
 * 修改备注：
 */
class BaseBindingViewHolder<Binding : ViewDataBinding>(view: View) : BaseViewHolder(view) {
    var binding: Binding? = null
}
